version: '3.8'
services:
  hainz-db:
    container_name: hainz-db
    image: postgres:15.1-alpine
    restart: always
    environment:
      - POSTGRES_USER=${USERNAME}
      - POSTGRES_PASSWORD=${PASSWORD}
      - POSTGRES_DB=${DB}
    healthcheck:
      test: ["CMD", "pg_isready", "-h", "localhost", "-p", "${PORT}", "-U", "${USERNAME}"]
      interval: 1s
      timeout: 3s
      retries: 5
    ports:
      - 5432:${PORT}
    volumes:
      - db:/var/lib/postgresql/data
    networks:
      - hainz-network
  hainz-app:
    container_name: hainz-app
    image: hainz-app
    restart: always
    build: .
    depends_on:
      hainz-db:
        condition: service_healthy
    networks:
      - hainz-network
    environment:
      - PERSISTENCE_USERNAME=${USERNAME}
      - PERSISTENCE_PASSWORD=${PASSWORD}
      - PERSISTENCE_HOSTNAME=${HOSTNAME}
      - PERSISTENCE_PORT=${PORT}
      - PERSISTENCE_DB=${DB}

volumes:
  db:
    driver: local

networks:
  hainz-network:
    driver: bridge